extractor/src/main/java/org/schabi/newpipe/extractor/stream/StreamInfo.java

```diff
@@ -163,13 +163,21 @@ private static StreamInfo extractStreams(StreamInfo streamInfo, StreamExtractor
         Exception dashMpdError = null;
         if (!isNullOrEmpty(streamInfo.getDashMpdUrl())) {
             try {
-                DashMpdParser.ParserResult result = DashMpdParser.getStreams(streamInfo);
-                streamInfo.getVideoOnlyStreams().addAll(result.getVideoOnlyStreams());
-                streamInfo.getAudioStreams().addAll(result.getAudioStreams());
-                streamInfo.getVideoStreams().addAll(result.getVideoStreams());
-                streamInfo.segmentedVideoOnlyStreams = result.getSegmentedVideoOnlyStreams();
-                streamInfo.segmentedAudioStreams = result.getSegmentedAudioStreams();
-                streamInfo.segmentedVideoStreams = result.getSegmentedVideoStreams();
+                final List<VideoStream> videoStreams =
+                        new ArrayList<>(streamInfo.getVideoStreams());
+                final List<VideoStream> videoOnlyStreams =
+                        new ArrayList<>(streamInfo.getVideoOnlyStreams());
+                final List<AudioStream> audioStreams =
+                        new ArrayList<>(streamInfo.getAudioStreams());
+
+                final DashMpdParser.Result result = DashMpdParser.getStreams(streamInfo);
```

Comment:
How about we call it `YoutubeDashMpdParser` and move this code to `YoutubeStreamExtractor`? Let the service extractors do dash manifest parsing if required.